File: Graph/C#/1260 DFS와 BFS.cs
using System;
using System.Collections.Generic;

class Program
{
    static List<List<int>> graph = new List<List<int>>();
    static bool[] visited = new bool[1001];
    static int n, m, v;
    
    static void DFS(int start) {
        Stack<int> stack = new Stack<int>();
        stack.Push(start);
        Console.Write("{0} ", start);
        visited[start] = true;
        
        while(stack.Count != 0) {
            int v = stack.Peek();
            bool cantmove = false;
            
            foreach(int n in graph[v]) {
                if(visited[n] == false) {
                    visited[n] = true;
                    stack.Push(n);
                    Console.Write("{0} ", n);
                    cantmove = true;
                    break;
                }
            }
            
            if(!cantmove) stack.Pop();
            
        }
    }
    static void BFS(int start) {
        Queue<int> queue = new Queue<int>();
        visited[start] = true;
        queue.Enqueue(start);
        Console.Write("{0} ", start);
        
        while(queue.Count != 0) {
            int v = queue.Dequeue();
            foreach(int n in graph[v]) {
                if(visited[n] == false) {
                    visited[n] = true;
                    queue.Enqueue(n);
                    Console.Write("{0} ", n);
                }
            }
        }
    }
    
    
    static void Main() {
        string s = Console.ReadLine();
        string[] sa = s.Split(' ');
        n = int.Parse(sa[0]); m = int.Parse(sa[1]); v = int.Parse(sa[2]);
        for(int i = 0 ; i < n+1; i++) {
            graph.Add(new List<int>());
        }
        for(int i = 0 ; i < m; i++) {
            string sm = Console.ReadLine();
            string[] sma = sm.Split(' ');
            int start = int.Parse(sma[0]);
            int end = int.Parse(sma[1]);
            
            graph[start].Add(end);
            graph[end].Add(start);
        }
        
        foreach(var info in graph) info.Sort();
        
        DFS(v);
        Array.Fill<bool>(visited, false);
        Console.WriteLine();
        BFS(v);
    }
}

File: Graph/Kotlin/1260 DFS와 BFS.kt
import java.util.Scanner
import java.util.*


var n: Int = 0
var m: Int = 0
var v: Int = 0
var graph = Array<ArrayList<Int>>(1001) {ArrayList<Int>()}
var visited = Array<Boolean>(1001) { false };


fun bfs(start : Int) {
    var queue = LinkedList<Int>()
    queue.add(start); visited[start] = true
    print("${start} ")
    
    
    while(queue.isEmpty() == false) {
        val v = queue.peek();
        queue.poll();
        for(x in graph[v]) {if(visited[x] == false) {queue.add(x); print("${x} "); visited[x] = true}}
    }
}

fun dfs(v : Int) {
    if(visited[v]) return
    
    visited[v] = true
    print("${v} ")
    for(n in graph[v]) dfs(n)
}


fun init() {
    visited.fill(false, 0, visited.size)
}


fun main(args: Array<String>) {
    var sc = Scanner(System.`in`)
    n = sc.nextInt()
    m = sc.nextInt()
    v = sc.nextInt()
    
    for(i in 0 until m) {
        var from = sc.nextInt(); val to = sc.nextInt()
        graph[from].add(to); graph[to].add(from)
    }
    
    for(i in 0 until graph.size) {graph[i].sort()}
    
    dfs(v)
    init()
    bfs(v)

}
